packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -28,10 +28,49 @@ export function PreviewIndicator({ children }) {
 }
 
 let buildId
+let pageData
```

Comment:
I think it'd be better to store this with `useState` instead of a global

Reply:
I did try that, but it did not work propertly.

Reply:
This is what I had mentioned where for whatever reason `useState` wasn't working as intended, which is why `buildId` is also global state. I had paired with you @TylerBarnes and Matt on separate occasions a while back trying to figure this out but no one could haha